### ISSUE TITLE
Support link commands in the output

### ISSUE THREAD

**Issue by rizsotto** [rizsotto/Bear#276]
There is a documented use case for this feature immunant/c2rust#227

**Comment by rizsotto** [comment]
Hey @PolyProgrammist, thanks for your comment.

Until the official definition of the [JSON compilation database](https://clang.llvm.org/docs/JSONCompilationDatabase.html) does not accommodate the link commands explicitly, I would emit the link command in a backward compatible way. (The [c2rust](https://github.com/immunant/c2rust/issues/227#issuecomment-574379588) solution looks like a good candidate for that.)

If Bear would emit a new format, I would like to call that something else, but not "JSON compilation database" to avoid confusion.

If Bear would emit a new format, I would like to onboard different build steps not only compilation and linking. My suggestion would be to describe the build in an ontology. (The file format can be anything like turtle or n3.) An ontology is a flexible knowledge representation, which can be extended easily, so can contain not only compilation or link commands, but which source file contains what function declaration, or what header files the source file includes, or even the call graph can be described in it. That could be a much detailed description of a software project. And would open possibilities for new use cases: can merge two dependent project's ontology without problem, and run cross project analysis. (That would catch bugs of new kind, when developers not using another project API correctly. And having fewer mono-repo project this can be relevant.) It can help to automate build system migrations, etc...

**Comment by KevinIsSun** [comment]
Hi @rizsotto , do we have a roadmap for this feature? Or do we have any design doc which can be public?  Thanks a lot.

**Comment by rizsotto** [comment]
@KevinIsSun are you asking about to implement the link command in the output as c2rust project suggest? The design doc is only the immunant/c2rust#227 ticket discussion. (Can expand that into a more detailed wiki page.) Are you interested to help with the implementation?

**Comment by KevinIsSun** [comment]
Yeah, I want to get link command in the output of bear. I have read the whole related discussion, but it's far away from a real feature.  I'm interested in the implementation, but I'm not familiar with the code of bear. I'm not sure how much time will spent on it. I think we need a design doc first.

**Comment by tr4v3ler** [comment]
Someone once implemented the capture of the link command on an old bear version. The link is as follows: [bear](https://github.com/Machiry/Bear)
Hope to provide some reference.

**Comment by tr4v3ler** [comment]
The following output is a case:
```
"arguments": [
    "-EL",
    "-r"
],
"directory": "/home/tr4v3ler/data/tools/difuze/mtk_kernel_0/kernel-3.18/out",
"files": [
    "drivers/misc/mediatek/video/common/mtkfb_fence.o", 
    "drivers/misc/mediatek/video/common/fbconfig_kdebug.o", 
    "drivers/misc/mediatek/video/common/aal20/built-in.o", 
    "drivers/misc/mediatek/video/common/corr10/built-in.o", 
    "drivers/misc/mediatek/video/common/color20/built-in.o"
],
"loader": "ld",  
"output": [
    "-o",
    "drivers/misc/mediatek/video/common/built-in.o"
]
```
There are many modules in the Linux kernel. I can't analyze the code of the whole module through a single object file, so I need to get the link command to generate the linked bitcode file through llvm and then complete the code analysis. But I'm not familiar with the source code of bear. I don't know where to modify it.
